src/PactNet/IMessageBuilder.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+
+namespace PactNet
+{
+    /// <summary>
+    /// Build up a mock message for a v3 pact
+    /// </summary>
+    public interface IPactMessageBuilderV3
+    {
+        /// <summary>
+        /// Add a new message to the pact
+        /// </summary>
+        /// <param name="description">Message description</param>
+        /// <returns>Fluent builder</returns>
+        IPactMessageBuilderV3 ExpectsToReceive(string description);
+
+        /// <summary>
+        /// Add a provider state
+        /// </summary>
+        /// <param name="providerState">Provider state description</param>
+        /// <returns>Fluent builder</returns>
+        IPactMessageBuilderV3 Given(string providerState);
```

Comment:
This API allows you to call the methods in an invalid order (e.g. you can call `WithMetadata` before you've called `ExpectsToReceive`. The request/response API separates out the interfaces and uses the type-state pattern so that you _must_ call `UponReceiving` first, and therefore you can't perform invalid sequences of operations.

Reply:
As discussed, this will be covered with the splitting of the builders.